components/support/nimbus-fml/src/cli.yaml

```diff
@@ -14,59 +14,131 @@ args:
         multiple: true
         help: Sets the level of verbosity
     - INPUT:
-        help: Sets the input file to use
-        required: true
+        # This will move to the generate command when deprecated commands are removed.
+        help: Sets the input file to use (deprecated, use `generate` command instead)
+        required: false
         index: 1
-    - ir:
-        help: The input file is intermediate representation. Useful for debugging FML.
-        long: ir
-        global: true
     - output:
-        help: The file where generated code is created
+        # This will move to the generate command when deprecated commands are removed.
+        help: The file where generated code is created (deprecated, use `generate` command instead)
         short: o
         long: output
         value_name: FILE
+        required: false
         global: true
     - channel:
-        help: The channel to generate the defaults for
+        # This will move to the generate command when deprecated commands are removed.
+        help: The release channel to generate code for
         long: channel
         short: h
         global: true
         takes_value: true
 subcommands:
+    - intermediate-repr:
+        about: Generate the intermediate representation json, useful for debugging
+    - generate:
+        about: Generate feature structs against the Feature Variables API.
+        args:
+            - INPUT:
+                help: Sets the input file to use
+                required: true
+                index: 1
+            - OUTPUT:
+                help: The file or directory where generated code is created
+                required: true
+                index: 2
+            - language:
+                help: The language of the output file
+                long: language
+                takes_value: true
+                possible_values:
+                  - swift
+                  - kotlin
+            #  Uncomment this once we remove the global option above.
+            # - channel:
+            #     help: The channel to generate the defaults for
+            #     long: channel
+            #     global: false
+            #     takes_value: true
+
+            # These next two flags aren't wired up yet.
+            # This will be done in EXP-2573.
+            - cache-dir:
+                help: The directory where downloaded files are cached
```

Comment:
nit: could be nice to add a message in the help that those aren't wired up yet in case anyone is confused, maybe with a reference to the tickets